### PR TITLE
chore: Log execution time for git status operation

### LINKED ISSUE #24340: [Task]: Lightweight git status call for page-load git status calls
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

_No response_

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
This PR sends the execution time as an analytics event.

#### PR fixes following issue(s)
Fixes #24340

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [x] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 1]
```diff
@@ -1892,7 +1891,36 @@ private Mono<GitStatusDTO> getStatus(
...
+                                "repoUrl",
+                                app.getGitApplicationMetadata().getRemoteUrl(),
+                                "executionTime",
+                                elapsedTime);
+                        return analyticsService
+                                .sendEvent(
+                                        AnalyticsEvents.UNIT_EXECUTION_TIME.getEventName(),
+                                        currentUser.getUsername(),
+                                        data)
+                                .thenReturn(gitStatusDTO);
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:1922]: Can we move this to a new method? Also why do we need so many parameters? We are just trying to get the execution time right?

Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:1922]: @AnaghHegde Yeah, but @infinitetrooper suggested to keep it same as other UNIT_EXECUTION_TIME events.

Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:1922]: Cool. Moving this to a new method can help in utilising this later and keep the status method more clean.

**Review by AnaghHegde — APPROVED** [review]

**Review by Parthvi12 — APPROVED** [review]

**Pull request merged**